rotkehlchen/tests/api/test_nfts.py

```diff
@@ -176,3 +176,64 @@ def test_nft_balances_and_prices(rotkehlchen_api_server):
     assert expected_result == result
     assert FVal(price) > 0
     assert FVal(price_usd) > 0
+
+
+@flaky(max_runs=3, min_passes=1)  # all opensea calls have become quite flaky
+@pytest.mark.parametrize('ethereum_accounts', [[TEST_ACC4]])
+@pytest.mark.parametrize('start_with_valid_premium', [True])
+@pytest.mark.parametrize('ethereum_modules', [['nfts']])
+@pytest.mark.parametrize('endpoint', ['nftsbalanceresource', 'nftsresource'])
+def test_nfts_ignoring_works(rotkehlchen_api_server, endpoint):
+    """Check that ignoring NFTs work as expected"""
+    # ignore the nft
+    response = requests.put(
+        api_url_for(
+            rotkehlchen_api_server,
+            'ignoredassetsresource',
+        ),
+        json={'assets': [NFT_ID_FOR_TEST_ACC4]},
+    )
+    result = assert_proper_response_with_result(response)
+    assert NFT_ID_FOR_TEST_ACC4 in result
+
+    # query the nfts endpoint and verify that the nft is not present
+    response = requests.get(
+        api_url_for(
+            rotkehlchen_api_server,
+            endpoint,
+        ),
+        json={'async_query': False},
+    )
+    result = assert_proper_response_with_result(response)
+    if endpoint == 'nftsresource':
+        all_nfts_ids = {nft['token_identifier'] for nft in result['addresses'][TEST_ACC4]}
+        assert NFT_ID_FOR_TEST_ACC4 not in all_nfts_ids
+    else:
+        assert result == {}
+
+    # remove the nft from the ignored list.
+    response = requests.delete(
+        api_url_for(
+            rotkehlchen_api_server,
+            'ignoredassetsresource',
+        ),
+        json={'assets': [NFT_ID_FOR_TEST_ACC4]},
+    )
+    result = assert_proper_response_with_result(response)
+    assert NFT_ID_FOR_TEST_ACC4 not in result
+
+    # query the nfts endpoint again and verify that the nft is present
+    response = requests.get(
+        api_url_for(
+            rotkehlchen_api_server,
+            endpoint,
+        ),
+        json={'async_query': False, 'ignore_cache': True},
+    )
+    result = assert_proper_response_with_result(response)
+    if endpoint == 'nftsresource':
+        all_nfts_ids = {nft['token_identifier'] for nft in result['addresses'][TEST_ACC4]}
+        assert NFT_ID_FOR_TEST_ACC4 in all_nfts_ids
+    else:
+        all_nfts_ids = {nft['id'] for nft in result[TEST_ACC4]}
+        assert NFT_ID_FOR_TEST_ACC4 in all_nfts_ids
```

Comment:
The assert is common for both logics so it can be written outside the if/else statment